org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/uriresolver/CacheResourceDownloadedException.java

```diff
@@ -26,7 +26,7 @@ public class CacheResourceDownloadedException extends CacheResourceException {
 
 	public enum CacheResourceDownloadedError {
 
-		ERROR_WHILE_DOWNLOADING("Error while downloading ''{0}'' to ''{1}''.");
+		ERROR_WHILE_DOWNLOADING("Error while downloading ''{0}'' to ''{1}'' : ''{2}''.");

```

Comment:
If possible, it would be nice to get the HTTP status code put into this message so that the user has a starting point to start figuring out why the download failed (i.e. 404 could mean they put the schema address in wrong, 407 could mean they configured their proxy incorrectly)

Reply:
I'm not sure if I'm able to get the HTTP status code since we are using URLConnection https://github.com/eclipse/lemminx/blob/0535b585f54e5b22c9af532eb802bd27803c841c/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/uriresolver/CacheResourcesManager.java#L192

Reply:
We may not be able to get it in every case (ie. if a url connection isn't using http), but it should be possible to check whether the object we're treating as a URLConnection happens to be an HttpURLConnection. If so, the cast should succeed and error code should be there.

https://github.com/eclipse/lemminx/blob/0535b585f54e5b22c9af532eb802bd27803c841c/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/uriresolver/CacheResourcesManager.java#L253

Reply:
> We may not be able to get it in every case

An another usecase is when connection is refused, we cannot get the HTTP status.